include/pmacc/Environment.tpp

```diff
@@ -252,10 +256,10 @@ namespace pmacc
 
         void EnvironmentContext::setDevice(int deviceNumber)
         {
-            int num_gpus = 0; // number of gpus
-            cuplaGetDeviceCount(&num_gpus);
+            int num_gpus = manager::Device<ComputeDevice>::get().count();
```

Comment:
Just a minor naming suggestion as this also counts the cpu backend (as it should). With the current naming I initially thought that if there are no GPUs ``setDevice()`` would fail. Though with how slow the CI is currently, maybe putting it in the next PR is also acceptable. 
```suggestion
            int num_devices = manager::Device<ComputeDevice>::get().count();
```

Reply:
should be `numDevices` to follow our naming rules